### ISSUE TITLE
Pin 3.7 build issues

### ISSUE THREAD

**Issue by Jinmo** [sslab-gatech/qsym#9]
Currently QSYM is using Pin 2.14 build 71313, and it doesn't support latest linux kernel versions. I wonder why, and tried to build it with Pin 3.7, but there were some issues:

- Latest PinCRT uses memory.h but it's overriden by `./memory.h` on `pintool/`. Renaming it into `_memory.h` fixed the issue.
- seems like LLVM's APInt, APSInt uses C++11 standard, but Pin's latest approach is replacing all C++ headers into their own one (it's not C++ >= 11 compatible), maybe using another integer class or copying whole dependencies into `pintool/third_party/llvm` will work.

**Comment by jakkdu** [comment]
Does libz3.so work in PIN 3.7?

**Comment by Jinmo** [comment]
doesn't work with the [DT_HASH issue](https://stackoverflow.com/questions/28638809/android-ndk-unsatisfiedlinkerror-dlopen-failed-empty-missing-dt-hash).

Currently trying to build it with sysv, and static shared library..

**Comment by Jinmo** [comment]
This is gonna be hard, since pin's dlopen doesn't support hash-style=gnu, just supports hash-style=sysv. I had to recompile all the glibc ~ z3 chain with `LDFLAGS='-Wl,--hash-style=sysv'` to support this. (`--static-libc` fails, `--static-libgcc`/`--static-lib[std]c++` works, ...)

Maybe ubuntu packages are not built for this kind of usage. (building a static library)

Currently I've failed to build glibc locally because of binutils internal bug, maybe waiting for PinCRT updates or using other libc like musl is a better choice..
